kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/InternalCompletableFutureTest.java

```diff
@@ -56,6 +57,16 @@ void asyncChainingMethodExecutesOnThreadOfExecutor() throws Exception {
 
     }
 
+    // this is illustrating a limitation of our internal stages, they do not compose nicely because toCompletableFuture throws
+    // an exception. This is to discourage users from joining or otherwise blocking. However CompletableFuture often relies on it
+    // internally.
+    @Test
+    void canNotCompose() {
+        CompletionStage<Object> stage = InternalCompletableFuture.completedFuture(executor, null).minimalCompletionStage();
+        CompletableFuture<Object> thenCompose = CompletableFuture.completedFuture(null).thenCompose(f -> stage);
+        assertThat(thenCompose).failsWithin(Duration.ofSeconds(2));
+    }
```

Comment:
I wonder if the benefits of the guard rails are worth it if our futures don't compose.